src/main/java/uk/gov/ida/matchingserviceadapter/validators/validationrules/ValidationRule.java

```diff
@@ -3,17 +3,14 @@
 import java.util.function.Predicate;
 
 public abstract class ValidationRule<T> {
-    private Predicate<T> predicate;
 
-    protected ValidationRule(Predicate<T> predicate) {
-        this.predicate = predicate;
-    }
+    protected abstract Predicate<T> getPredicate();
 
     protected void apply(T subject) {
-        if (!predicate.test(subject)) {
+        if (!getPredicate().test(subject)) {
             throwException();
         }
     }
 
-    public abstract void throwException();
+    protected abstract void throwException();
```

Comment:
Actually, technically it would be better to have a method `protected abstract Throwable getException()` and throw it in the parent method as we cannot guarantee that a void method will actually throw....